tests/FeeSharingCollectorTest.js

```diff
@@ -291,21 +288,508 @@ contract("FeeSharingCollectorProxy:", (accounts) => {
         await sovryn.setMaxDisagreement(maxDisagreement);
 
         RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT =
-            await feeSharingCollectorProxy.RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT();
+            await feeSharingCollector.RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT();
 
         return sovryn;
     }
 
     beforeEach(async () => {
         await loadFixture(protocolDeploymentFixture);
     });
+    describe("withdrawStartingFromCheckpoint, withdrawRBTCStartingFromCheckpoint and getNextPositiveUserCheckpoint", () => {
+        let snapshot;
+        before(async () => {
+            await loadFixture(protocolDeploymentFixture);
+        });
+        beforeEach(async () => {
+            snapshot = await takeSnapshot();
+        });
+        afterEach(async () => {
+            await snapshot.restore();
```

Comment:
Good idea to introduce the snapshot here.
Might be useful for other test cases

Reply:
this is a workaround in this particular case 
`loadFixture` takes a snapshot and should revert to it when subsequently called  
but sometimes (and in this particular case too) it doesn't revert or not completely reverts for unclear reason causing the upfollowing tests to fail - they pass when run isolated though